### ISSUE TITLE
tensorflow 2.0 variable slice assign_add not supported

### ISSUE THREAD

**Issue by motionlife** [tensorflow/tensorflow#30468]
Could anybody tell me in tf 2.0 rc1, why is variable slice assignment (varaible[...].assign function) works well but the assign_add is not supported?
e.g. :

`var = tf.Variable(tf.ones([2,3,3]))`
`var[:, :, 1].assign(tf.zeros([2,3]))  # this will work`

`var[:, :, 1].assign_add(tf.zeros([2,3]))  # this will NOT work`
`AttributeError: 'tensorflow.python.framework.ops.EagerTensor' object has no attribute 'assign_add'`

**Comment by rmothukuru** [comment]
Reproduced this issue with TF Version 2.0.beta.

**Comment by ziofil** [comment]
@jaingaurav Any update on this issue?

**Comment by motionlife** [comment]
slice assign_add still not supported in tf 2.0 official release

**Comment by jaingaurav** [comment]
I haven't had a chance to make much progress on this issue, but it requires doing similar tricks we do here for `assign`: https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/ops/array_ops.py#L1091. We'll likely have to add other operators beyond just `assign_add`.

**Comment by ziofil** [comment]
bump

**Comment by jaingaurav** [comment]
ziofil@: This is definitely something we want to address as it is a usability issue. We're still trying to allocate time to work on this. In the meantime, contributions are welcome.

**Comment by gadagashwini** [comment]
Reproduce this issue with Tf 2.1 and Tf-nightly==2.2.0.dev20200312.
Please find the gist [here](https://colab.sandbox.google.com/gist/gadagashwini/c82696421987c70062a38cf2c5d95866/untitled452.ipynb). Thanks!

**Comment by zapaishchykova** [comment]
what I did is replacing this:
`total_recall = tf.assign_add(total_recall, current_recall)`
with
`total_recall = tf.compat.v1.assign_add(total_recall, current_recall)`

**Comment by sushreebarsa** [comment]
Was able to replicate the issue with TF v2.5,please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/110ab4428e1b60348012b20e27b630a6/30468-nightly.ipynb) ..Thanks!

**Comment by kumariko** [comment]
I could reproduce the issue with TF 2.6 . Please, find the gist [**`here`**](https://colab.research.google.com/gist/kumariko/da64ddd5e26c9b9f6a391e1b275aadc0/30468-nightly.ipynb#scrollTo=JjSEl8bUsizl).Thanks!

**Comment by gadagashwini** [comment]
Hi @motionlife, Looks like issue is resolved on Tf v2.9.0.

```
import tensorflow as tf

var = tf.Variable(tf.ones([2,3,3]))
res_vas_assign = var.assign(tf.zeros([2,3,3]))
res_var = var.assign_add(tf.zeros([2,3,3]))
res_var.numpy()
```

**Output**
```
array([[[0., 0., 0.],
        [0., 0., 0.],
        [0., 0., 0.]],

       [[0., 0., 0.],
        [0., 0., 0.],
        [0., 0., 0.]]], dtype=float32)
```
